scripts/aws/cloudformation/tasking-manager.template.js

```diff
@@ -133,296 +79,233 @@ const Parameters = {
   TaskingManagerLogo: {
     Description: "URL for logo",
     Type: "String"
-  }
+  },
+  SMTPPassword: {
+    Description: "Secrets Manager entry for SMTP Password",
+    Type: "String"
+  },
+  OAuth2ConsumerSecret: {
+    Description: "Secrets Manager entry for OAuth2 Consumer Secret",
+    Type: "String"
+  },
+  NewRelicLicenseKey: {
+    Description: "Secrets Manager entry for NewRelic License Key",
+    Type: "String"
+  },
+  ImageUploadAPIKey: {
+    Description: "Secrets Manager entry for Image Upload API Key",
+    Type: "String"
+  },
+  TaskingManagerManagedSecret: {
+    Description: "Secrets Manager entry for Tasking Manager Secret",
+    Type: "String"
+  },
 };
 
 const Conditions = {
-  UseASnapshot: cf.notEquals(cf.ref('DBSnapshot'), ''),
-  DatabaseDumpFileGiven: cf.notEquals(cf.ref('DatabaseDump'), ''),
-  IsTaskingManagerProduction: cf.equals(cf.ref('AutoscalingPolicy'), 'production'),
-  IsTaskingManagerDemo: cf.equals(cf.ref('AutoscalingPolicy'), 'Demo (max 3)'),
+  IsTaskingManagerProduction: cf.equals(cf.ref('DeploymentEnvironment'), 'production'),
+  IsTaskingManagerDemo: cf.equals(cf.ref('DeploymentEnvironment'), 'demo'),
   IsHOTOSMUrl: cf.equals(
     cf.select('1', cf.split('.', cf.ref('TaskingManagerURL')))
     , 'hotosm')
 };
 
+/*
+ * Resources
+ *
+ * ECS Cluster, ECS Definition, ECS Service
+ * Secrets Policy :- Secrets Manager Secrets
+ * ECSTaskRole :-
+ * ECSTaskExecutionRole
+ * TaskingManagerASG, TaskingManagerASGPolicy
+ * TaskingManager Load Balancer
+ * Load Balancer Route 53
+ * Target Group
+ * ALB HTTPS Listener, ALB HTTP Listener
+ * RDS Managed Password
+ * RDS Instance
+ * RDS Secret attachment
+ * React Bucket S3
+ * React Bucket Policy
+ * React CloudFront
+ * Route 53
+ *
+ */
 const Resources = {
-  TaskingManagerASG: {
-    DependsOn: 'TaskingManagerLaunchConfiguration',
-    Type: 'AWS::AutoScaling::AutoScalingGroup',
+  ECSCluster: {
+    Type: 'AWS::ECS::Cluster',
     Properties: {
-      AutoScalingGroupName: cf.stackName,
-      Cooldown: 300,
-      MinSize: cf.if('IsTaskingManagerProduction', 3, 1),
-      DesiredCapacity: cf.if('IsTaskingManagerProduction', 3, 1),
-      MaxSize: cf.if('IsTaskingManagerProduction', 9, cf.if('IsTaskingManagerDemo', 3, 1)),
-      HealthCheckGracePeriod: 600,
-      LaunchConfigurationName: cf.ref('TaskingManagerLaunchConfiguration'),
-      TargetGroupARNs: [ cf.ref('TaskingManagerTargetGroup') ],
-      HealthCheckType: 'EC2',
-      AvailabilityZones: ['us-east-1a', 'us-east-1b', 'us-east-1c', 'us-east-1d', 'us-east-1f'],
-      Tags: [{
-        Key: 'Name',
-        PropagateAtLaunch: true,
-        Value: cf.stackName
-      }]
-    },
-    UpdatePolicy: {
-      AutoScalingRollingUpdate: {
-        PauseTime: 'PT60M',
-        MaxBatchSize: 2,
-        WaitOnResourceSignals: true
-      }
+      CapacityProviders: [
+        'FARGATE',
+        'FARGATE_SPOT'
+      ],
+      ClusterName: cf.stackName, // TODO: Change this
```

Comment:
Does this need to be changed?

Reply:
Not at the moment. We can do this later as it's a cosmetic change.